test/typescript/iterators/main.ts

```diff
@@ -0,0 +1,172 @@
+import { Construct } from "constructs";
+import {
+  App,
+  TerraformStack,
+  Testing,
+  TerraformVariable,
+  TerraformIterator,
+  Fn,
+  TerraformOutput,
+  TerraformAsset,
+  AssetType,
+} from "cdktf";
+import * as ArchiveProvider from "./.gen/providers/archive";
+import * as NomadProvider from "./.gen/providers/nomad";
+
+export class TestIterators extends TerraformStack {
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+
+    new ArchiveProvider.ArchiveProvider(this, "archive");
+
+    const list = new TerraformVariable(this, "files", {
+      type: "list(string)",
+      default: ["a", "b"],
+    });
+
+    const it = TerraformIterator.fromList(list.listValue);
+
+    const files = new TerraformAsset(this, "files-dir", {
+      path: "./files",
+      type: AssetType.DIRECTORY,
+    });
+
+    // TODO: we don't support using .dynamic for simple mapping (i.e. to a string instead of to an object) yet
```

Comment:
Do we throw in these cases?

Reply:
No, currently only our types would prohibit this. And I don't think we need to throw, as it would already work to pass a string or number from a technical perspective. It just is not intended yet and not tested.

Reply:
created and linked